pgdog/src/backend/pool/address.rs

```diff
@@ -48,9 +50,9 @@ impl Address {
         }
     }
 
-    /// Get address for `TCPStream`.
-    pub fn addr(&self) -> String {
-        format!("{}:{}", self.host, self.port)
+    pub async fn addr(&self) -> Result<SocketAddr, std::io::Error> {
```

Comment:
Check out the `backend::Error` enum. We use `thiserror` for returning errors most places, it might have an `std::io::Error` mapping already and you can continue to use  `?`.